Frends.CertificateStore/Frends.CertificateStore.cs

```diff
@@ -1,51 +1,179 @@
-﻿using System.ComponentModel;
+﻿using System;
+using System.Collections.Generic;
```

Comment:
Now most of these tasks will never actually fail. The user has to always check success and if not throw an exception. 
Could it be  better to actually just allow the tasks to fail? Perhaps add an option to "Do not throw on error" for the task configuration?